src/core/globalstate.ts

```diff
@@ -45,6 +45,12 @@ export class MobXGlobals {
 	isRunningReactions = false;
 
 	/**
+	 * Current reaction (if any is running).
+	 * Reactions are run after derivations using a trampoline.
+	 */
+	currentReaction: Reaction = undefined;
```

Comment:
@mweststrate, reaction's `effect` function is not running inside derivation, so if we use looking into `derivationStack` we can't dispose reaction like it was suggested:
```js
reaction(
  () => myStore.data,
  data => {
    if (data) {
      doSomethingWithData(data);
      disposeCurrentReaction();
    }     
  }
)
```

Insead we'll have to do it like this (may be less user-friendly, I think):

```js
reaction(
  () => {
    const data = myStore.data;
    if (data) disposeCurrentReaction();
    return data;
  },
  data => {
    if (data) {
      doSomethingWithData(data);
    }     
  }
)
```

We have 3 options:
1. Pop derivation from `derivationStack` _after_ effect is called (I'm not sure that is ideologically correct)
2. Introduce a state var `currentReaction` (currently in this PR), and remove state var `isRunningReactions` in favour of `currentReaction`.
3. Accept that `disposeCurrentReaction` must be called inside reaction's expression.

Which way you think is better?